src/utils/packageHasher.js

```diff
@@ -0,0 +1,52 @@
+const logger = require('../Logger');
+const fs = require('fs');
+const md5File = require('md5-file');
+const walkedFiles = new Map();
+const hashMap = new Map();
+const Path = require('path');
+
+
+async function packageHasher(mainBundle, detailed = false) {
+  walkBundles(mainBundle, '--');
+}
+
+module.exports = packageHasher;
+
+async function walkBundles(bundle, indent) {
+  for (let child of bundle.childBundles) {
+    if ( !walkedFiles.has(child.name)) {
+      walkBundles(child, indent + '--')
+    }
+  }
+
+  renameReferences(bundle.name, hashMap);
+
+  let oldHash = Path.basename(bundle.name, Path.extname(bundle.name));
+  let hash = md5File.sync(bundle.name);
+  walkedFiles.set(bundle.name, hash);
+  hashMap.set(oldHash, hash);
+  // logger.log(indent + bundle.name)
+
+  let newLocation = `${bundle.entryAsset.options.outDir}/${hash}.${bundle.entryAsset.type}`;
+  // logger.log(indent + newLocation + " (move here)");
+
+  if (bundle.parentBundle) {
+    fs.renameSync(bundle.name, newLocation);
+  }
+}
+
+async function renameReferences(fileName, hashMap) {
+  if (fs.existsSync(fileName)) {
+    data = fs.readFileSync(fileName, 'utf8')
+
+    hashMap.forEach(function(value, key) {
+      data = data.replace(new RegExp(key, 'g'), value);
```

Comment:
This seems somewhat dangerous and would probably end up replacing things we don't want. For example, if you had a file called `index.js` but some string in the app also contained that but wasn't a reference, we'd end up replacing it too.

Instead of doing it this way, maybe we can store a list of reference locations as part of the asset (e.g. part of the options to `addDependency`)?

Reply:
@devongovett in my limited experience, `bundle.name` has only ever been my entry file (`entry.html` in my case) or an md5 hash. What are the other possibilities?

Reply:
@devongovett in my testing, `hashMap` only contains md5 hashes of things. Are there other non-md5 names that can get assigned to a bundle?

Reply:
@devongovett okay....

If a name isn't hashed in `generateBundleName()`, would we want this new code to hash it? What's the reason for leaving it as-is there and not just hashing everything?

Reply:
My point is mostly that doing a search and replace over output files for filenames is somewhat dangerous in that we might end up replacing things we didn't intend. Perhaps `generateBundleName` could generate something random always which is much less likely to appear in the source code, and we could use that to replace with the final names here.

Reply:
@devongovett yeah, if it's safe for `generateBundleName` to _always_ use hashing, that would be nice.